examples/pinocchio/src/qap.rs

```diff
@@ -0,0 +1,362 @@
+use crate::{common::*, r1cs::R1CS};
+use lambdaworks_math::polynomial::Polynomial;
+use std::convert::From;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+/// QAP Representation of the circuits
+pub struct QuadraticArithmeticProgram {
+    pub vs: Vec<Polynomial<FE>>,
+    pub ws: Vec<Polynomial<FE>>,
+    pub ys: Vec<Polynomial<FE>>,
+    pub target: Polynomial<FE>,
+    pub number_of_inputs: usize,
+    pub number_of_outputs: usize,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum CreationError {
+    PolynomialVectorsSizeMismatch,
+}
+
+impl QuadraticArithmeticProgram {
+    /// Creates a new QAP
+    /// This expects vectors to be organized like:
+    /// v0,w0,y0
+    /// inputs associated v,w,y polynomials
+    /// mid associated polynomials
+    /// outputs associated v,w,y polynomials
+    pub fn new(
+        vs: Vec<Polynomial<FE>>,
+        ws: Vec<Polynomial<FE>>,
+        ys: Vec<Polynomial<FE>>,
+        target: Polynomial<FE>,
+        number_of_inputs: usize,
+        number_of_outputs: usize,
+    ) -> Result<Self, CreationError> {
+        if vs.len() != ws.len() || vs.len() != ys.len() || ws.len() != ys.len() {
+            Err(CreationError::PolynomialVectorsSizeMismatch)
+        } else {
+            Ok(Self {
+                vs,
+                ws,
+                ys,
+                target,
+                number_of_inputs,
+                number_of_outputs,
```

Comment:
If I understand correctly, there's also a condition that `number_of_inputs + number_of_outputs <= vs.len()`, is that correct?